onnxruntime/core/providers/webnn/builders/impl/max_min_op_builder.cc

```diff
@@ -87,25 +87,32 @@ bool MaxMinOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& /* initializ
   return true;
 }
 
-bool MaxMinOpBuilder::HasSupportedInputsImpl(const Node& node, const WebnnDeviceType /* device_type */,
+bool MaxMinOpBuilder::HasSupportedInputsImpl(const Node& node, const emscripten::val& wnn_limits,
                                              const logging::Logger& logger) const {
   const auto& input_defs = node.InputDefs();
   const auto& op_type = node.OpType();
   int32_t input0_type;
   int32_t input1_type;
 
-  if (!GetType(*input_defs[0], input0_type, logger) ||
-      !GetType(*input_defs[1], input1_type, logger))
+  if (!GetType(*input_defs[0], input0_type, logger))
     return false;
 
-  if (!IsSupportedDataType(input0_type, webnn_supported_data_types)) {
+  if (input_defs.size() > 1 && !GetType(*input_defs[1], input1_type, logger)) {
+    return false;
+  }
+
+  std::string webnn_op_type;
+  if (!GetWebNNOpType(op_type, webnn_op_type))
+    return false;
+
+  if (!IsSupportedDataType(input0_type, wnn_limits[webnn_op_type]["a"]["dataTypes"])) {
     LOGS(logger, VERBOSE) << "[" << op_type
                           << "] Input type: [" << input0_type
                           << "] is not supported for now";
     return false;
   }
 
-  if (input0_type != input1_type) {
+  if (input_defs.size() > 1 && input0_type != input1_type) {
```

Comment:
I've noticed `input0_type != input1_type` a few times too, often enough that having a data type comparison and logging function could be useful. 🤷‍♂️

Reply:
OK, I will create a common util.